gef.py

```diff
@@ -6082,9 +6110,79 @@ def pprint_bin(arena_addr, index, _type=""):
             fw = chunk.fwd
             nb_chunk += 1
 
-        gef_print("  ".join(m))
+        if m:
+            gef_print("  ".join(m))
         return nb_chunk
 
+
+@register_command
+class GlibcHeapTcachebinsCommand(GenericCommand):
+    """Display information on the Tcachebins on an arena (default: main_arena).
+    See https://sourceware.org/git/?p=glibc.git;a=commitdiff;h=d5c3fafc4307c9b7a4c7d5cb381fcdbfad340bcc."""
+
+    _cmdline_ = "heap bins tcache"
+    _syntax_  = "{:s} [ARENA_ADDRESS]".format(_cmdline_)
+
+    def __init__(self):
+        super(GlibcHeapTcachebinsCommand, self).__init__(complete=gdb.COMPLETE_LOCATION)
+        return
+
+    @only_if_gdb_running
+    def do_invoke(self, argv):
+        # Determine if we are using libc with tcache built in (2.26+)
+        libc = gdb.execute("vmmap libc", to_string=True)
```

Comment:
Why not use `get_process_maps()` ?

Reply:
Yes, much better. Though luckily me doing it this way made me discover #362 :)

Reply:
See bf6bec9.

It's a bit longer, but probably better that one extra round of generating and then re-parsing a string.